plugins/org.yakindu.sct.doc.user/src/user-guide/c-domain.textile

```diff
@@ -255,6 +252,138 @@ p. With this definition we can use expressions like @t.a.x@, see the image below
+
+bc. pTriangle2 = rotateTriangle(pTriangle, pCenterPoint, 45.0);
+
+bq.. *Please note:
+
+Assigning a pointer to a pointer variable is only possible if the pointer types are the same.
+
+
+h2(#cdom_arrays). Arrays
+
+Unlike other variables, arrays are not defined in a statechart's definition section, but rather on the C side in header files. Importing a C header containing an array definition makes the array available to a statechart.
+
+While YAKINDU Statechart Tool's Deep C Integration provides mechanisms for accessing individual elements of an existent array, arrays must be allocated statically or dynamically in C. Initializing the array elements is possible in C as well as in the statechart. However, depending on the concrete application it might generally be easier in C.
+
+The header file _sample_arrays.h_ defines a couple of sample arrays:
+
+bc.. #ifndef SAMPLE_ARRAYS_H_
+#define SAMPLE_ARRAYS_H_
+
+#include <stdint.h>
+#include "triangle.h"
+
+int32_t coordinates[] = {0, 0, 10, 0, 5, 5};
+
+Triangle manyTriangles[200];
+
+int32_t * pArray[10];
+
+#endif /* SAMPLE_ARRAYS_H_ */
+
+p. The following arrays are defined:
+* _coordinates_ is statically allocated to hold six _int32_t_ elements and it is initialized with values for all six of them. More precisely, the number of elements in the initializer determines the size of the array.
+* _manyTriangles_ is statically allocated with enough memory to hold 200 elements of the self-defined _Triangle_ type. However, these elements are not initialized. This can and should be done either in C or in the state machine. An example is given below.
+* _pArray_ is of size 10 and holds pointers to _int32_t_ values.
+
+As mentioned above, importing a header file containing array definitions into the statechart's definition section is sufficient to make the arrays available in a statechart. Example:
+
+bc. import: sample_arrays
+
+With this import, you can access the arrays in statechart language expressions, for example in a state's local reactions:
+
+bc. entry /
+coordinates[2] = 42
+
+Writing to array elements is as straightforward as you would expect. Examples:
+
+bc. coordinates[0] = coordinates[0] + 1;
+pArray[3] = n.pointer;
+pArray[4] = coordinates[0].pointer
+
+Passing arrays as parameters to C functions is straightforward. Let's say you have a C function _sort_ to sort the elements of a one-dimensional _int32_t_ array and return a pointer to the sorted array:
+
+bc. int32_t* sort(int32_t data[], int size) {…}
+
+Please note that in C a function cannot return an array as such, but only a pointer to it. Analogously you cannot pass an array as such as a parameter to a function, but only a pointer to that array. To express this in the function's parameter type, you could specify the _sort_ function equivalently as follows. The _data_ parameter is now specified as @int32_t* data@ instead of @int32_t data[]@, but the meaning is the same.
```

Comment:
The array __is__ a pointer, pointing to the first element. declaring a function with `int32_t data[]` as argument is valid and the user might be confused here.